src/nix-build/nix-build.cc

```diff
@@ -387,6 +387,13 @@ static void main_nix_build(int argc, char * * argv)
 
         if (dryRun) return;
 
+        if (settings.isExperimentalFeatureEnabled("ca-derivations")) {
+            auto resolvedDrv = drv.tryResolve(*store);
+            if (!resolvedDrv)
+                throw Error("unable to resolve the derivation '%s'. nix-shell can’t continue", drvInfo.queryDrvPath());
```

Comment:
Should we do
```suggestion
            assert(resolvedDrv);
```
?

I think it should always resolve because the previous step builds all the deps, right?

Reply:
It should, yes. But given that this isn’t in any way performance sensitive and that `assert` makes for some pretty ugly errors, I’d rather keep the nice exception

Reply:
Fair. Maybe some sort of "internal error" and/or "this should be impossible because we just resolved" then?

Reply:
I eventually put an assert, just with a dirty hack to add some slightly more meaningful message to the error

(Makes me think that it would be nice to have a standard way of having assert with error messages)